// Generated by FunAPI app_proto_generator.  DO NOT EDIT!
//
// source: giving_tree_client_messages.proto
//         giving_tree_server_messages.proto


import "account_messages.proto";
import "giving_tree_client_messages.proto";
import "giving_tree_server_messages.proto";


message ClientAppMessageType {
  enum Type {
    kPlayerRegisterName = 1001;
    kPlayerTakeApple = 1002;
  }
}


extend ClientAppMessage {
  optional ClientAppMessageType.Type client_message_type = 1000;

  optional PlayerRegisterName player_register_name = 1001;
  optional PlayerTakeApple player_take_apple = 1002;
}


message ServerAppMessageType {
  enum Type {
    kTreeCountDown = 1001;
    kGameResult = 1002;
  }
}


extend ServerAppMessage {
  optional ServerAppMessageType.Type server_message_type = 1000;

  optional TreeCountDown tree_count_down = 1001;
  optional GameResult game_result = 1002;
}


